WordPress/src/main/java/org/wordpress/android/widgets/WPPrefView.java

```diff
@@ -0,0 +1,425 @@
+
+                setPrefType(PrefType.fromInt(prefTypeInt));
+                setHeading(heading);
+                setTitle(title);
+                setSummary(summary);
+                setTextDialogSubtitle(dialogSubtitle);
+                setShowDivider(showDivider);
+            } finally {
+                a.recycle();
+            }
+        }
+    }
+
+    public void setOnPrefChangedListener(OnPrefChangedListener listener) {
+        mListener = listener;
+    }
+
+    private void doPrefChanged() {
+        if (mListener != null) {
+            mListener.onPrefChanged(this);
+        }
+    }
+
+    private void setPrefType(@NonNull PrefType prefType) {
+        mPrefType = prefType;
+
+        boolean isToggle = mPrefType == PrefType.TOGGLE;
+        mTitleTextView.setVisibility(isToggle ? View.GONE : View.VISIBLE);
+        mToggleSwitch.setVisibility(isToggle ? View.VISIBLE : View.GONE);
+
+        if (isToggle) {
+            mContainer.setOnClickListener(null);
+            mToggleSwitch.setOnCheckedChangeListener(this);
+        } else {
+            mContainer.setOnClickListener(this);
+            mToggleSwitch.setOnCheckedChangeListener(null);
+        }
+    }
+
+    /*
+     * blue heading text that should appear above the preference when it's the first in a group
+     */
+    private void setHeading(String heading) {
+        mHeadingTextView.setText(heading);
+        mHeadingTextView.setVisibility(TextUtils.isEmpty(heading) ? GONE : VISIBLE);
+    }
+
+    /*
+     * title above the preference and below the optional heading
+     */
+    private void setTitle(String title) {
+        mTitleTextView.setText(title);
+        mToggleSwitch.setText(title);
+    }
+
+    /*
+     * optional description that appears below the title
+     */
+    public void setSummary(String summary) {
+        mSummaryTextView.setText(summary);
+        mSummaryTextView.setVisibility(TextUtils.isEmpty(summary) ? View.GONE : View.VISIBLE);
+    }
+
+    /*
+     * subtitle on the dialog that appears when the PrefType is TEXT
+     */
+    private void setTextDialogSubtitle(String subtitle) {
+        mTextDialogSubtitle = subtitle;
+    }
+
+    /*
+     * current entry when the PrefType is TEXT
+     */
+    public String getTextEntry() {
+        return mTextEntry;
+    }
+    public void setTextEntry(String entry) {
+        mTextEntry = entry;
+        setSummary(entry);
+    }
+
+    /*
+     * returns whether or not the switch is checked when the PrefType is TOGGLE
+     */
+    public boolean isChecked() {
+        return mPrefType == PrefType.TOGGLE && mToggleSwitch.isChecked();
+    }
+    public void setChecked(boolean checked) {
+        mToggleSwitch.setChecked(checked);
+    }
+
+    /*
+     * determines whether a divider appears below the pref - should be true for all but the last one
+     */
+    private void setShowDivider(boolean show) {
```

Comment:
We're only calling this from `initView`, I think we should just inline it there since it's not doing anything complex.